hydra-plutus/src/Hydra/Contract/Head.hs

```diff
@@ -299,9 +301,25 @@ checkClose ctx headContext parties initialUtxoHash snapshotNumber closedUtxoHash
 -- | Checks that the datum
 makeContestationDeadline :: ContestationPeriod -> ScriptContext -> POSIXTime
 makeContestationDeadline cperiod ScriptContext{scriptContextTxInfo} =
-  case ivTo (txInfoValidRange scriptContextTxInfo) of
-    UpperBound (Finite time) _ -> addContestationPeriod time cperiod
-    _ -> traceError "no upper bound validaty interval defined for close"
+  let txValidFrom = ivFrom (txInfoValidRange scriptContextTxInfo)
+      txValidTo = ivTo (txInfoValidRange scriptContextTxInfo)
+   in case (txValidFrom, txValidTo) of
+        (LowerBound (Finite startTime) _, UpperBound (Finite endTime) _) ->
+          -- calculate new upper bound by adding the upper bound to the contestation period
+          let newUpperBound = addContestationPeriod endTime cperiod
+              -- FIXME: what do we do here? We need to say contest deadline is within
+              -- the reasonable bounds (so not 10 years in the future) but from where
+              -- do we get this value? We have access to the contest deadline but not a
+              -- way of saying that this contest deadline is reasonable.
+              -- One way could be to add contestationDeadline to the Open constructor too?
```

Comment:
@ch1bo this is my main open question

Reply:
I think it's rather that we want to check that the upper bound is not too far away from the lower bound. We do not want to depend on how big the contestation period is.

So basically we want to check:

```
withinBounds = endTime - startTime < someAcceptableLimit 
```

Annoyingly we do need to set the limit as `POSIXTime`, but I think we can hard-code something. The difference (and also the "grace time" off-chain) exists because we need some time until the transaction reaches the block producer / diffuses through the network.